src/samplers/cpu/cpi/mod.rs

```diff
@@ -0,0 +1,91 @@
+use super::stats::*;
+use super::*;
+use crate::common::Counter;
+use perf_event::events::Hardware;
+use perf_event::{Builder, Group};
+
+#[distributed_slice(CPU_SAMPLERS)]
+fn init(config: &Config) -> Box<dyn Sampler> {
+    Box::new(Cpi::new(config))
+}
+
+pub struct Cpi {
+    prev: Instant,
```

Comment:
the use of `prev` `next` and the general ticking mechanism is used repeatedly, I think we may want to refactor that in provide a trait that's shared.